pallet/staking/src/lib.rs

```diff
@@ -1315,44 +1313,42 @@ where
 			state_mutability: StateMutability::Payable,
 		};
 
-		let notify_transaction = LegacyTransaction {
-			nonce: U256::zero(),     // Will be reset in the message transact call
-			gas_price: U256::zero(), // Will be reset in the message transact call
-			gas_limit: U256::from(1_000_000), /* It should be big enough for the evm
-			                          * transaction, otherwise it will out of gas. */
+		let notify_transaction = EIP1559Transaction {
+			chain_id: 0,                            // Will be reset in the message transact call
+			nonce: U256::zero(),                    // Will be reset in the message transact call
+			max_fee_per_gas: U256::zero(),          // Will be reset in the message transact call
+			max_priority_fee_per_gas: U256::zero(), // Will be reset in the message transact call
+			gas_limit: U256::from(1_000_000),       /* It should be big enough for the evm
+			                                         * transaction, otherwise it will out of gas. */
 			action: TransactionAction::Call(reward_distr),
 			value: U256::zero(),
 			input: function
 				.encode_input(&[Token::Address(staking_reward), Token::Uint(amount.into())])
 				.unwrap_or_default(),
-			signature,
+			access_list: vec![],
+			// copied from:
+			// https://github.com/rust-ethereum/ethereum/blob/24739cc8ba6e9d8ee30ada8ec92161e4c48d578e/src/transaction.rs#L873-L875
+			odd_y_parity: false,
+			// 36b241b061a36a32ab7fe86c7aa9eb592dd59018cd0443adc0903590c16b02b0
+			r: H256([
+				54, 178, 65, 176, 97, 163, 106, 50, 171, 127, 232, 108, 122, 169, 235, 89, 45, 213,
+				144, 24, 205, 4, 67, 173, 192, 144, 53, 144, 193, 107, 2, 176,
+			]),
+			// 5edcc541b4741c5cc6dd347c5ed9577ef293a62787b4510465fadbfe39ee4094
+			s: H256([
+				54, 178, 65, 176, 97, 163, 106, 50, 171, 127, 232, 108, 122, 169, 235, 89, 45, 213,
+				144, 24, 205, 4, 67, 173, 192, 144, 53, 144, 193, 107, 2, 176,
+			]),
 		};
 		// b"sc/ktstk"
 		let sender =
 			H160([115, 99, 47, 107, 116, 115, 116, 107, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
 
 		if let Err(e) = <darwinia_message_transact::Pallet<T>>::message_transact(
 			LcmpEthOrigin::MessageTransact(sender).into(),
```

Comment:
```suggestion
			Sub2EthOrigin::MessageTransact(sender).into(),
```

Reply:
- pallet name: `RuntimeTransact`
- origin name: `RuntimeEthOrigin`
- function name: `runtime_transact(xxx)`

What do you think of the renaming above? 

`sub2`, the substrate is renamed to `polkadot-vm`, so I think we should avoid introducing something like `subxxx` anymore ~ `runtime` is a better word for me. The meaning is also clear enough.

Reply:
LGTM